src/Fantomas.Tests/CommentTests.fs

```diff
@@ -86,6 +86,42 @@ let print_30_permut () =
     permutation
 """
 
+[<Test>]
+let ``comments on let bindings with return type, 2043`` () =
+    formatSourceString
+        false
+        """
+let count: int [] // foo
+    = [2]
+    """
+        config
+    |> prepend newline
+    |> should
+        equal
+        """
+let count: int [] = [ 2 ] // foo
```

Comment:
This isn't correct I'm afraid.
It should be something like:
```fsharp
let count: int [] // foo
    =
     [2]
```
The comment now is placed after the body expression.
It should be after the return type.